### ISSUE TITLE
Builder pattern

### ISSUE THREAD

**Issue by pnrobinson** [TheJacksonLaboratory/biodownload#1]
The basic structure could work something like this

```
Biodownloader downloader = BiodownloaderBuilder.create()
  .hpoJson()
  .goJson()
  .goGaf()
  .medgene2MIM()
  .custom("my-local-file", "my-valid-url")
  .build();
List<File> downloadedFiles = downloader.download();
// or
List<File> downloadedFiles = downloader.overwriteDownload();
```
We would need to decide what to do if one or more of the downloads did not work. It is probably best just to throw a runtime exception, since the user will not be able to proceed (usually) unless all files are downloaded.

By default if a file is already there, the File path is returned and a message is sent to the logger that the file download is beign skipped. There is an additional method to download and overwrite existing files (what is the best name for this?).

@belkassaby @ielis

**Comment by belkassaby** [comment]
Hi @pnrobinson @ielis , I've just pushed a branch with fixes for issues #1 to #10 . Let me know what you think of it. I also changed the name and folder of the project to BioDownload.

**Comment by belkassaby** [comment]
I also changed the name of the project and folder to BioDownload.
